encoding/json/decode.go

```diff
@@ -333,12 +355,19 @@ func (d *Decoder) decodeInt64(valueJSON interface{}) cadence.Int64 {
 		panic(ErrInvalidJSONCadence)
 	}
 
-	return cadence.NewInt64(i)
+	return cadence.NewInt64(d.gauge, i)
 }
 
 func (d *Decoder) decodeInt128(valueJSON interface{}) cadence.Int128 {
 	bigInt := d.decodeBigInt(valueJSON)
```

Comment:
The current code is a good start, in the future we actually also need to meter allocations here: The JSON string is converted to a bigint value, which already allocates, which should go into the function below:
https://github.com/onflow/cadence/pull/1611/files#diff-486c5071ade706b28d51beacaf8b7cbe4267a892e708877e4ef37e3cfbc464d4R367.

We can address this in a follow-up PR though, not a blocker for this one

Reply:
I went ahead and did this change since I needed to fix up the PR anyway.